react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState('');
+  const [valid, setValid] = useState(false);
```

Comment:
This can be calculated from `password`, so the best approach is to _not_ have it as state. Instead, declare a local `valid` variable and set it according to the calculations you do.